compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4179,6 +4180,49 @@ declare_lint! {
     "named arguments in format used positionally"
 }
 
+declare_lint! {
+    /// The `never_type_fallback_flowing_into_unsafe` lint detects cases where never type fallback
```

Comment:
the lint description should mention what never type fallback actually is, I think most people are lot aware of it

Reply:
That's one of the annoying problems of all the never type stuff, fallback can't be explained without going somewhat indepth :(

Reply:
I added an explanation, does it look good?